### DISCUSSION TITLE
Best way to share models with frontend application

### DISCUSSION THREAD

**Discussion by pinkynrg** [adonisjs/core#3993] [Help]
I was wondering the best approach to share models' types with the frontend.

I was thinking to export a models.ts inside the contracts folder and write it as follows:

```
export type IMyModel = ModelAttributes<MyModel>
// ... the rest of the models 
```

this would be shared as an npm package, but since i would use Lerna it would be easier.

Since it is the first time i share typescript BE models to the  FE i was wondering if this made any sense. 

Maybe there is a builtin way to do this, so I wanted to make sure.

**Comment by stereobooster** [comment]
Related https://github.com/adonisjs/core/discussions/4017
